<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>miaViz • miaViz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="miaViz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">miaViz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/miaViz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>miaViz</h1>
            
      

      <div class="hidden name"><code>miaViz.Rmd</code></div>

    </div>

    
    
<p><code>miaViz</code> implements plotting function to work with
<code>TreeSummarizedExperiment</code> and related objects in a context
of microbiome analysis. For more general plotting function on
<code>SummarizedExperiment</code> objects the <code>scater</code>
package offers several options, such as <code>plotColData</code>,
<code>plotExpression</code> and <code>plotRowData</code>.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install <code>miaViz</code>, install <code>BiocManager</code>
first, if it is not installed. Afterwards use the <code>install</code>
function from <code>BiocManager</code> and load <code>miaViz</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"miaViz"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaViz</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="abundance-plotting">Abundance plotting<a class="anchor" aria-label="anchor" href="#abundance-plotting"></a>
</h2>
<p>In contrast to other fields of sequencing based fields of research
for which expression of genes is usually studied, microbiome research
uses the more term <sub>Abundance</sub> to described the numeric data
measured and analyzed. Technically, especially in context of
<code>SummarizedExperiment</code> objects, there is no difference.
Therefore <code>plotExpression</code> can be used to plot
<code>Abundance</code> data of a particular feature.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html" class="external-link">plotExpression</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, </span>
<span>              features <span class="op">=</span> <span class="st">"549322"</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-2-1.png" width="3000"></p>
<p>On the other hand, plotAbundance can be used to plot abundance by
<code>rank</code>. A bar plot is returned showing the relative abundance
within each sample for a given <code>rank</code>. At the same time the
<code>features</code> argument can be set to <code>NULL</code>
(default).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GlobalPatterns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Kingdom"</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-4-1.png" width="3000"></p>
<p>If <code>rank</code> is set to null however then the bars will be
colored by each individual taxon. Please note that if you’re doing this
make sure to agglomerate your data to a certain taxonomic hand before
plotting.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GlobalPatterns_king</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/agglomerate-methods.html" class="external-link">agglomerateByRank</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, <span class="st">"Kingdom"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns_king</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-5-1.png" width="3000"></p>
<p>With subsetting to selected features the plot can be fine tuned.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prev_phylum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/getPrevalence.html" class="external-link">getPrevalent</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>                                detection <span class="op">=</span> <span class="fl">0.01</span>, onRankOnly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span><span class="op">$</span><span class="va">Phylum</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">prev_phylum</span><span class="op">]</span>,</span>
<span>              rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>              assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-7-1.png" width="3000"></p>
<p>The <code>features</code> argument is reused for plotting data along
the different samples. In the next example the <sub>SampleType</sub> is
plotted along the samples. In this case the result is a list, which can
combined using external tools, for example <code>patchwork</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span><span class="op">$</span><span class="va">Phylum</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">prev_phylum</span><span class="op">]</span>,</span>
<span>                       features <span class="op">=</span> <span class="st">"SampleType"</span>,</span>
<span>                       rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>                       assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="va">plots</span><span class="op">$</span><span class="va">abundance</span> <span class="op">/</span> <span class="va">plots</span><span class="op">$</span><span class="va">SampleType</span> <span class="op">+</span></span>
<span>     <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">9</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-8-1.png" width="3000"></p>
<p>Further example about composition barplot can be found at
Orchestrating Microbiome Analysis <span class="citation">(Lahti, Shetty,
and Ernst 2021)</span>.</p>
</div>
<div class="section level2">
<h2 id="prevalence-plotting">Prevalence plotting<a class="anchor" aria-label="anchor" href="#prevalence-plotting"></a>
</h2>
<p>To visualize prevalence within the dataset, two functions are
available, <code>plotFeaturePrevalence</code>,
<code>plotPrevalenceAbundance</code> and
<code>plotPrevalence</code>.</p>
<p><code>plotFeaturePrevalence</code> produces a so-called landscape
plot, which visualizes the prevalence of samples across abundance
thresholds.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotPrevalence.html">plotFeaturePrevalence</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>                   detections <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.001</span>, <span class="fl">0.01</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in plotFeaturePrevalence(GlobalPatterns, rank = "Phylum", detections =</span></span>
<span><span class="co">#&gt; c(0, : The 'plotFeaturePrevalence' function is deprecated. Use</span></span>
<span><span class="co">#&gt; 'plotRowPrevalence' instead.</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-9-1.png" width="3000"></p>
<p><code>plotPrevalenceAbundance</code> plot the prevalence depending on
the mean relative abundance on the chosen taxonomic level.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotPrevalence.html">plotPrevalentAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Family"</span>,</span>
<span>                       colour_by <span class="op">=</span> <span class="st">"Phylum"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-10-1.png" width="3000"></p>
<p><code>plotPrevalence</code> plot the number of samples and their
prevalence across different abundance thresholds. Abundance steps can be
adjusted using the <code>detections</code> argument, whereas the
analyzed prevalence steps is set using the <code>prevalences</code>
argument.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotPrevalence.html">plotPrevalence</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,</span>
<span>               rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>               detections <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="fl">0.1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">5</span>, <span class="fl">10</span>, <span class="fl">20</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>,</span>
<span>               prevalences <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">1</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-11-1.png" width="3000"></p>
</div>
<div class="section level2">
<h2 id="tree-plotting">Tree plotting<a class="anchor" aria-label="anchor" href="#tree-plotting"></a>
</h2>
<p>The information stored in the <code>rowTree</code> can be directly
plotted. However, sizes of stored trees have to be kept in mind and
plotting of large trees rarely makes sense.</p>
<p>For this example we limit the information plotted to the top 100 taxa
as judged by mean abundance on the genus level.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/agglomerate-methods.html" class="external-link">agglomerateByRank</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html" class="external-link">addPerFeatureQC</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">log_mean</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">mean</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span> <span class="op">/</span> <span class="fl">100</span></span>
<span><span class="va">top_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/summaries.html" class="external-link">getTop</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span>,</span>
<span>                       method<span class="op">=</span><span class="st">"mean"</span>,</span>
<span>                       top<span class="op">=</span><span class="fl">100L</span>,</span>
<span>                       assay.type<span class="op">=</span><span class="st">"counts"</span><span class="op">)</span></span></code></pre></div>
<p>Colour, size and shape of tree tips and nodes can be decorated based
on data present in the <code>SE</code> object or by providing additional
information via the <code>other_fields</code> argument. Note that
currently information for nodes have to be provided via the
<code>other_fields</code> arguments.</p>
<p>Data will be matched via the <code>node</code> or <code>label</code>
argument depending on which was provided. <code>label</code> takes
precedent.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,</span>
<span>            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span>,</span>
<span>            tip_size_by <span class="op">=</span> <span class="st">"detected"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot1-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size)" width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance
(colour) and prevalence (size)
</p>
</div>
<p>Tip and node labels can be shown as well. Setting
<code>show_label = TRUE</code> shows the tip labels only …</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,</span>
<span>            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span>,</span>
<span>            tip_size_by <span class="op">=</span> <span class="st">"detected"</span>,</span>
<span>            show_label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot2-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size). Tip labels of the tree are shown as well." width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance
(colour) and prevalence (size). Tip labels of the tree are shown as
well.
</p>
</div>
<p>… whereas node labels can be selectively shown by providing a named
logical vector to <code>show_label</code>.</p>
<p>Please note that currently <code>ggtree</code> can only plot node
labels in a rectangular layout.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Genus:Providencia"</span>, <span class="st">"Genus:Morganella"</span>, <span class="st">"0.961.60"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,</span>
<span>            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span>,</span>
<span>            tip_size_by <span class="op">=</span> <span class="st">"detected"</span>,</span>
<span>            show_label <span class="op">=</span> <span class="va">labels</span>,</span>
<span>            layout<span class="op">=</span><span class="st">"rectangular"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot3-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size). Selected node and tip labels are shown." width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance
(colour) and prevalence (size). Selected node and tip labels are shown.
</p>
</div>
<p>Information can also be visualized on the edges of the tree plot.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,</span>
<span>            edge_colour_by <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot4-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and edges labeled Kingdom (colour) and prevalence (size)" width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance
(colour) and edges labeled Kingdom (colour) and prevalence (size)
</p>
</div>
</div>
<div class="section level2">
<h2 id="graph-plotting">Graph plotting<a class="anchor" aria-label="anchor" href="#graph-plotting"></a>
</h2>
<p>Similar to tree data, graph data can also be plotted in conjunction
with <code>SummarizedExperiment</code> objects. Since the graph data in
itself cannot be stored in a specialized slot, a graph object can be
provided separately or as an element from the <code>metedata</code>.</p>
<p>Here we load an example graph. As graph data, all objects types
accepted by <code>as_tbl_graph</code> from the <code>tidygraph</code>
package are supported.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">col_graph</span><span class="op">)</span></span></code></pre></div>
<p>In the following examples, the <code>weight</code> data is
automatically generated from the graph data. The
<code>SummarizedExperiment</code> provided is required to have
overlapping rownames with the node names of the graph. Using this link
the graph plot can incorporate data from the
<code>SummarizedExperiment</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotGraph.html">plotColGraph</a></span><span class="op">(</span><span class="va">col_graph</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span>,</span>
<span>             colour_by <span class="op">=</span> <span class="st">"SampleType"</span>,</span>
<span>             edge_colour_by <span class="op">=</span> <span class="st">"weight"</span>,</span>
<span>             edge_width_by <span class="op">=</span> <span class="st">"weight"</span>,</span>
<span>             show_label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-15-1.png" width="3000"></p>
<p>As mentioned the graph data can be provided from the
<code>metadata</code> of the <code>SummarizedExperiment</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">metadata</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">graph</span> <span class="op">&lt;-</span> <span class="va">col_graph</span></span></code></pre></div>
<p>This produces the same plot as shown above.</p>
</div>
<div class="section level2">
<h2 id="plotting-of-serial-data">Plotting of serial data<a class="anchor" aria-label="anchor" href="#plotting-of-serial-data"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"miaTime"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"microbiome/miaTime"</span>, upgrade <span class="op">=</span> <span class="st">"never"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load data from miaTime package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"miaTime"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SilvermanAGutData</span>, package<span class="op">=</span><span class="st">"miaTime"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">SilvermanAGutData</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="va">taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/summaries.html" class="external-link">getTop</a></span><span class="op">(</span><span class="va">tse</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p>Data from samples collected along time can be visualized using
<code>plotSeries</code>. The <code>x</code> argument is used to
reference data from the <code>colData</code> to use as descriptor for
ordering the data. The <code>y</code> argument selects the feature to
show. Since plotting a lot of features is not advised a maximum of 20
features can plotted at the same time.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSeries.html">plotSeries</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>           x <span class="op">=</span> <span class="st">"DAY_ORDER"</span>,</span>
<span>           y <span class="op">=</span> <span class="va">taxa</span>,</span>
<span>           colour_by <span class="op">=</span> <span class="st">"Family"</span><span class="op">)</span></span></code></pre></div>
<p>If replicated data is present, data is automatically used for
calculation of the <code>mean</code> and <code>sd</code> and plotted as
a range. Data from different assays can be used for plotting via the
<code>assay.type</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSeries.html">plotSeries</a></span><span class="op">(</span><span class="va">tse</span><span class="op">[</span><span class="va">taxa</span>,<span class="op">]</span>,</span>
<span>           x <span class="op">=</span> <span class="st">"DAY_ORDER"</span>,</span>
<span>           colour_by <span class="op">=</span> <span class="st">"Family"</span>,</span>
<span>           linetype_by <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>           assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<p>Additional variables can be used to modify line type aesthetics.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSeries.html">plotSeries</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>           x <span class="op">=</span> <span class="st">"DAY_ORDER"</span>,</span>
<span>           y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/summaries.html" class="external-link">getTop</a></span><span class="op">(</span><span class="va">tse</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>           colour_by <span class="op">=</span> <span class="st">"Family"</span>,</span>
<span>           linetype_by <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>           assay.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plotting-factor-data">Plotting factor data<a class="anchor" aria-label="anchor" href="#plotting-factor-data"></a>
</h2>
<p>To visualize the relative relations between two groupings among the
factor data, two functions are available for the purpose;
<code>plotColTile</code> and <code>plotRowTile</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, package<span class="op">=</span><span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span>
<span><span class="fu"><a href="../reference/plotColTile.html">plotColTile</a></span><span class="op">(</span><span class="va">se</span>,<span class="st">"SampleType"</span>,<span class="st">"Primer"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x.top <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-23-1.png" width="3000"></p>
</div>
<div class="section level2">
<h2 id="dmn-fit-plotting">DMN fit plotting<a class="anchor" aria-label="anchor" href="#dmn-fit-plotting"></a>
</h2>
<p>Searching for groups that are similar to each other among the
samples, could be approached with the Dirichlet Multinomial Mixtures
<span class="citation">(Holmes, Harris, and Quince 2012)</span>. After
using <code>runDMN</code> from the <code>mia</code> package, several k
values as a number of clusters are used to observe the best fit (see
also <code>getDMN</code> and <code>getBestDMNFit</code>). To visualize
the fit using e.g. “laplace” as a measure of goodness of fit:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dmn_se</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">dmn_se</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "DMN"</span></span>
<span><span class="co"># plot the fit</span></span>
<span><span class="fu"><a href="../reference/plotDMN.html">plotDMNFit</a></span><span class="op">(</span><span class="va">dmn_se</span>, type <span class="op">=</span> <span class="st">"laplace"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .local(x, name, type, ...): 'getDMN' is deprecated.</span></span>
<span><span class="co">#&gt; Use 'addCluster' instead.</span></span>
<span><span class="co">#&gt; See help("Deprecated") and help("Now runDMN and calculateDMN are deprecated. Use addCluster with DMMParam parameter and full parameter set as true instead.-deprecated").</span></span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-24-1.png" width="3000"></p>
</div>
<div class="section level2">
<h2 id="serial-data-ordination-and-trajectories">Serial data ordination and trajectories<a class="anchor" aria-label="anchor" href="#serial-data-ordination-and-trajectories"></a>
</h2>
<p>Principal Coordinates Analysis using Bray-Curtis dissimilarity on the
<code>hitchip1006</code> dataset:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaTime</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hitchip1006</span>, package <span class="op">=</span> <span class="st">"miaTime"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">hitchip1006</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="co">## Ordination with PCoA with Bray-Curtis dissimilarity</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span><span class="va">tse</span>, FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>, method <span class="op">=</span> <span class="st">"bray"</span>, name <span class="op">=</span> <span class="st">"PCoA_BC"</span>,</span>
<span>              assay.type <span class="op">=</span> <span class="st">"relabundance"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># plot</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">tse</span>, dimred <span class="op">=</span> <span class="st">"PCoA_BC"</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<p>Retrieving information about all available trajectories:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># List subjects with two time points</span></span>
<span><span class="va">selected.subjects</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tse</span><span class="op">$</span><span class="va">subject</span><span class="op">)</span><span class="op">==</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Subjects counts per number of time points available in the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tse</span><span class="op">$</span><span class="va">subject</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>Timepoints<span class="op">=</span><span class="va">Var1</span>, Subjects<span class="op">=</span><span class="va">Freq</span><span class="op">)</span></span></code></pre></div>
<p>Lets look at all trajectories having two time points in the data:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot</span></span>
<span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_path</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">X1</span>, y<span class="op">=</span><span class="va">X2</span>, group<span class="op">=</span><span class="va">subject</span><span class="op">)</span>, </span>
<span>              arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">arrow</a></span><span class="op">(</span>length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="st">"inches"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>              <span class="co"># combining ordination data and metadata then selecting the subjects</span></span>
<span>              <span class="co"># Note, scuttle::makePerCellDF could also be used for the purpose.</span></span>
<span>              data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                            <span class="va">subject</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">selected.subjects</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">time</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"All trajectories with two time points"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Filtering the two time point trajectories by divergence and
displaying top 10%:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaTime</span><span class="op">)</span></span>
<span><span class="co"># calculating step wise divergence based on the microbial profiles</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">getStepwiseDivergence</span><span class="op">(</span><span class="va">tse</span>, group <span class="op">=</span> <span class="st">"subject"</span>, time_field <span class="op">=</span> <span class="st">"time"</span><span class="op">)</span></span>
<span><span class="co"># retrieving the top 10% divergent subjects having two time points</span></span>
<span><span class="va">top.selected.subjects</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                            <span class="va">subject</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">selected.subjects</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_frac</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="va">time_divergence</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">subject</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co"># plot</span></span>
<span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_path</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">X1</span>, y<span class="op">=</span><span class="va">X2</span>,</span>
<span>                  color<span class="op">=</span><span class="va">time_divergence</span>, group<span class="op">=</span><span class="va">subject</span><span class="op">)</span>,</span>
<span>              <span class="co"># the data is sorted in descending order in terms of time</span></span>
<span>              <span class="co"># since geom_path will use the first occurring observation</span></span>
<span>              <span class="co"># to color the corresponding segment. Without the sorting</span></span>
<span>              <span class="co"># geom_path will pick up NA values (corresponding to initial time</span></span>
<span>              <span class="co"># points); breaking the example.</span></span>
<span>              data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                            <span class="va">subject</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">top.selected.subjects</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>                  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">time</span><span class="op">)</span><span class="op">)</span>,</span>
<span>              <span class="co"># arrow end is reversed, due to the earlier sorting.</span></span>
<span>              arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">arrow</a></span><span class="op">(</span>length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="st">"inches"</span><span class="op">)</span>, ends <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Top 10%  divergent trajectories from time point one to two"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradient2</a></span><span class="op">(</span>low<span class="op">=</span><span class="st">"white"</span>, high<span class="op">=</span><span class="st">"red"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Plotting an example of the trajectory with the maximum total
divergence:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get subject with the maximum total divergence</span></span>
<span><span class="va">selected.subject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">subject</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>total_divergence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">time_divergence</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">total_divergence</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">total_divergence</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">subject</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co"># plot</span></span>
<span><span class="va">p</span> <span class="op">+</span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_path</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">X1</span>, y<span class="op">=</span><span class="va">X2</span>, group<span class="op">=</span><span class="va">subject</span><span class="op">)</span>,</span>
<span>              data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                            <span class="va">subject</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">selected.subject</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">time</span><span class="op">)</span>,</span>
<span>              arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">arrow</a></span><span class="op">(</span>length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="st">"inches"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Longest trajectory by divergence"</span><span class="op">)</span><span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>More examples and materials are available at Orchestrating Microbiome
Analysis <span class="citation">(Lahti, Shetty, and Ernst
2021)</span>.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] patchwork_1.2.0                 scater_1.33.2                  </span></span>
<span><span class="co">#&gt;  [3] scuttle_1.15.1                  miaViz_1.13.8                  </span></span>
<span><span class="co">#&gt;  [5] ggraph_2.2.1                    ggplot2_3.5.1                  </span></span>
<span><span class="co">#&gt;  [7] mia_1.13.27                     MultiAssayExperiment_1.31.4    </span></span>
<span><span class="co">#&gt;  [9] TreeSummarizedExperiment_2.13.0 Biostrings_2.73.1              </span></span>
<span><span class="co">#&gt; [11] XVector_0.45.0                  SingleCellExperiment_1.27.2    </span></span>
<span><span class="co">#&gt; [13] SummarizedExperiment_1.35.1     Biobase_2.65.0                 </span></span>
<span><span class="co">#&gt; [15] GenomicRanges_1.57.1            GenomeInfoDb_1.41.1            </span></span>
<span><span class="co">#&gt; [17] IRanges_2.39.1                  S4Vectors_0.43.1               </span></span>
<span><span class="co">#&gt; [19] BiocGenerics_0.51.0             MatrixGenerics_1.17.0          </span></span>
<span><span class="co">#&gt; [21] matrixStats_1.3.0               BiocStyle_2.33.1               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.4.1               ggplotify_0.1.2            </span></span>
<span><span class="co">#&gt;   [3] tibble_3.2.1                polyclip_1.10-6            </span></span>
<span><span class="co">#&gt;   [5] rpart_4.1.23                DirichletMultinomial_1.47.0</span></span>
<span><span class="co">#&gt;   [7] lifecycle_1.0.4             lattice_0.22-6             </span></span>
<span><span class="co">#&gt;   [9] MASS_7.3-61                 backports_1.5.0            </span></span>
<span><span class="co">#&gt;  [11] magrittr_2.0.3              Hmisc_5.1-3                </span></span>
<span><span class="co">#&gt;  [13] sass_0.4.9                  rmarkdown_2.27             </span></span>
<span><span class="co">#&gt;  [15] jquerylib_0.1.4             yaml_2.3.9                 </span></span>
<span><span class="co">#&gt;  [17] RColorBrewer_1.1-3          DBI_1.2.3                  </span></span>
<span><span class="co">#&gt;  [19] minqa_1.2.7                 abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [21] zlibbioc_1.51.1             purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [23] yulab.utils_0.1.4           nnet_7.3-19                </span></span>
<span><span class="co">#&gt;  [25] tweenr_2.0.3                sandwich_3.1-0             </span></span>
<span><span class="co">#&gt;  [27] GenomeInfoDbData_1.2.12     ggrepel_0.9.5              </span></span>
<span><span class="co">#&gt;  [29] irlba_2.3.5.1               tidytree_0.4.6             </span></span>
<span><span class="co">#&gt;  [31] vegan_2.6-6.1               rbiom_1.0.3                </span></span>
<span><span class="co">#&gt;  [33] pkgdown_2.1.0               permute_0.9-7              </span></span>
<span><span class="co">#&gt;  [35] DelayedMatrixStats_1.27.1   codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [37] DelayedArray_0.31.7         ggforce_0.4.2              </span></span>
<span><span class="co">#&gt;  [39] tidyselect_1.2.1            aplot_0.2.3                </span></span>
<span><span class="co">#&gt;  [41] UCSC.utils_1.1.0            farver_2.1.2               </span></span>
<span><span class="co">#&gt;  [43] lme4_1.1-35.5               ScaledMatrix_1.13.0        </span></span>
<span><span class="co">#&gt;  [45] viridis_0.6.5               base64enc_0.1-3            </span></span>
<span><span class="co">#&gt;  [47] jsonlite_1.8.8              BiocNeighbors_1.23.0       </span></span>
<span><span class="co">#&gt;  [49] decontam_1.25.0             tidygraph_1.3.1            </span></span>
<span><span class="co">#&gt;  [51] Formula_1.2-5               systemfonts_1.1.0          </span></span>
<span><span class="co">#&gt;  [53] tools_4.4.1                 ggnewscale_0.4.10          </span></span>
<span><span class="co">#&gt;  [55] treeio_1.29.0               ragg_1.3.2                 </span></span>
<span><span class="co">#&gt;  [57] Rcpp_1.0.12                 glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [59] gridExtra_2.3               SparseArray_1.5.20         </span></span>
<span><span class="co">#&gt;  [61] xfun_0.45                   mgcv_1.9-1                 </span></span>
<span><span class="co">#&gt;  [63] dplyr_1.1.4                 withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [65] BiocManager_1.30.23         fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [67] boot_1.3-30                 bluster_1.15.0             </span></span>
<span><span class="co">#&gt;  [69] fansi_1.0.6                 digest_0.6.36              </span></span>
<span><span class="co">#&gt;  [71] rsvd_1.0.5                  R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [73] gridGraphics_0.5-1          textshaping_0.4.0          </span></span>
<span><span class="co">#&gt;  [75] colorspace_2.1-0            lpSolve_5.6.20             </span></span>
<span><span class="co">#&gt;  [77] utf8_1.2.4                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [79] generics_0.1.3              data.table_1.15.4          </span></span>
<span><span class="co">#&gt;  [81] DECIPHER_3.1.4              graphlayouts_1.1.1         </span></span>
<span><span class="co">#&gt;  [83] httr_1.4.7                  htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [85] S4Arrays_1.5.4              pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [87] gtable_0.3.5                htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [89] bookdown_0.40               scales_1.3.0               </span></span>
<span><span class="co">#&gt;  [91] ggfun_0.1.5                 knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [93] rstudioapi_0.16.0           reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [95] checkmate_2.3.1             nlme_3.1-165               </span></span>
<span><span class="co">#&gt;  [97] nloptr_2.1.1                cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [99] zoo_1.8-12                  stringr_1.5.1              </span></span>
<span><span class="co">#&gt; [101] parallel_4.4.1              vipor_0.4.7                </span></span>
<span><span class="co">#&gt; [103] foreign_0.8-87              desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [105] pillar_1.9.0                grid_4.4.1                 </span></span>
<span><span class="co">#&gt; [107] vctrs_0.6.5                 slam_0.1-50                </span></span>
<span><span class="co">#&gt; [109] BiocSingular_1.21.2         beachmat_2.21.4            </span></span>
<span><span class="co">#&gt; [111] cluster_2.1.6               beeswarm_0.4.0             </span></span>
<span><span class="co">#&gt; [113] htmlTable_2.4.2             evaluate_0.24.0            </span></span>
<span><span class="co">#&gt; [115] mvtnorm_1.2-5               cli_3.6.3                  </span></span>
<span><span class="co">#&gt; [117] compiler_4.4.1              rlang_1.1.4                </span></span>
<span><span class="co">#&gt; [119] crayon_1.5.3                labeling_0.4.3             </span></span>
<span><span class="co">#&gt; [121] mediation_4.5.0             plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [123] fs_1.6.4                    ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt; [125] stringi_1.8.4               viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [127] BiocParallel_1.39.0         munsell_0.5.1              </span></span>
<span><span class="co">#&gt; [129] lazyeval_0.2.2              Matrix_1.7-0               </span></span>
<span><span class="co">#&gt; [131] sparseMatrixStats_1.17.2    highr_0.11                 </span></span>
<span><span class="co">#&gt; [133] igraph_2.0.3                memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [135] RcppParallel_5.1.8          bslib_0.7.0                </span></span>
<span><span class="co">#&gt; [137] ggtree_3.13.0               ape_5.8</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-DMM" class="csl-entry">
Holmes, Ian, Keith Harris, and Christopher Quince. 2012.
<span>“Dirichlet Multinomial Mixtures: Generative Models for Microbial
Metagenomics.”</span> <em>PLOS ONE</em>. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030126" class="external-link">https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030126</a>.
</div>
<div id="ref-OMA" class="csl-entry">
Lahti, Leo, Sudarshan Shetty, and Felix GM Ernst. 2021.
<span>“Orchestrating Microbiome Analysis.”</span> <a href="https://microbiome.github.io/OMA/" class="external-link">https://microbiome.github.io/OMA/</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Tuomas Borman, Felix G.M. Ernst, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
