<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>miaViz • miaViz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="miaViz">
<meta property="og:description" content="miaViz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">miaViz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/miaViz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="miaViz_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>miaViz</h1>
            
      
      
      <div class="hidden name"><code>miaViz.Rmd</code></div>

    </div>

    
    
<p><code>miaViz</code> implements plotting function to work with <code>TreeSummarizedExperiment</code> and related objects in a context of microbiome analysis. For more general plotting function on <code>SummarizedExperiment</code> objects the <code>scater</code> package offers several options, such as <code>plotColData</code>, <code>plotExpression</code> and <code>plotRowData</code>.</p>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install <code>miaViz</code>, install <code>BiocManager</code> first, if it is not installed. Afterwards use the <code>install</code> function from <code>BiocManager</code> and load <code>miaViz</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"miaViz"</span><span class="op">)</span>
<span class="co">#&gt; Bioconductor version 3.13 (BiocManager 1.30.10), R Under development (unstable)</span>
<span class="co">#&gt;   (2021-03-18 r80099)</span>
<span class="co">#&gt; Installing package(s) 'miaViz'</span>
<span class="co">#&gt; Warning: package 'miaViz' is not available for this version of R</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A version of this package for your version of R might be available elsewhere,</span>
<span class="co">#&gt; see the ideas at</span>
<span class="co">#&gt; https://cran.r-project.org/doc/manuals/r-devel/R-admin.html#Installing-packages</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">miaViz</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></code></pre></div>
</div>
<div id="abundance-plotting" class="section level1">
<h1 class="hasAnchor">
<a href="#abundance-plotting" class="anchor"></a>Abundance plotting</h1>
<p>in contrast to other fields of sequencing based fields of research for which expression of genes is usually studied, microbiome research uses the more term <sub>Abundance</sub> to described the numeric data measured and analyzed. Technically, especially in context of <code>SummarizedExperiment</code> objects, there is no difference. Therefore <code>plotExpression</code> can be used to plot <code>Abundance</code> data.</p>
<p><code>plotAbundance</code> can be used as well and as long as <code>rank</code> is set <code>NULL</code>, it behaves as <code>plotExpression</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="cn">NULL</span>, 
              features <span class="op">=</span> <span class="st">"549322"</span>, abund_values <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-3-1.png" width="3000"></p>
<p>However, if the <code>rank</code> is set not <code>NULL</code> a bar plot is returned. At the same time the <code>features</code> argument can be set to <code>NULL</code> (default).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GlobalPatterns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformCounts.html">transformCounts</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Kingdom"</span>, abund_values <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-5-1.png" width="3000"></p>
<p>With subsetting to selected features the plot can be fine tuned.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prev_phylum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/getPrevalence.html">getPrevalentTaxa</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Phylum"</span>,
                                detection <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">[</span><span class="fu">rowData</span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span><span class="op">$</span><span class="va">Phylum</span> <span class="op">%in%</span> <span class="va">prev_phylum</span><span class="op">]</span>,
              rank <span class="op">=</span> <span class="st">"Phylum"</span>,
              abund_values <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-7-1.png" width="3000"></p>
<p>The <code>features</code> argument is reused for plotting data along the different samples. In the next example the <sub>SampleType</sub> is plotted along the samples. In this case the result is a list, which can combined using external tools, for example <code>patchwork</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>
<span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotAbundance.html">plotAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">[</span><span class="fu">rowData</span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span><span class="op">$</span><span class="va">Phylum</span> <span class="op">%in%</span> <span class="va">prev_phylum</span><span class="op">]</span>,
                       features <span class="op">=</span> <span class="st">"SampleType"</span>,
                       rank <span class="op">=</span> <span class="st">"Phylum"</span>,
                       abund_values <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span>
<span class="va">plots</span><span class="op">$</span><span class="va">abundance</span> <span class="op">/</span> <span class="va">plots</span><span class="op">$</span><span class="va">SampleType</span> <span class="op">+</span>
     <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html">plot_layout</a></span><span class="op">(</span>heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">9</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-8-1.png" width="3000"></p>
</div>
<div id="prevalence-plotting" class="section level1">
<h1 class="hasAnchor">
<a href="#prevalence-plotting" class="anchor"></a>Prevalence plotting</h1>
<p>To visualize prevalence within the dataset, two functions are available, <code>plotTaxaPrevalence</code>, <code>plotPrevalenceAbundance</code> and <code>plotPrevalence</code>.</p>
<p><code>plotTaxaPrevalence</code> produces a so-called landscape plot, which visualizes the prevalence of samples across abundance thresholds.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotPrevalence.html">plotTaxaPrevalence</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Phylum"</span>,
                   detections <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.001</span>, <span class="fl">0.01</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-9-1.png" width="3000"></p>
<p><code>plotPrevalenceAbundance</code> plot the prevalence depending on the mean relative abundance on the chosen taxonomic level.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotPrevalence.html">plotPrevalentAbundance</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, rank <span class="op">=</span> <span class="st">"Family"</span>,
                       colour_by <span class="op">=</span> <span class="st">"Phylum"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-10-1.png" width="3000"></p>
<p><code>plotPrevalence</code> plot the number of samples and their prevalence across different abundance thresholds. Abundance steps can be adjusted using the <code>detections</code> argument, whereas the analyzed prevalence steps is set using the <code>prevalences</code> argument.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotPrevalence.html">plotPrevalence</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,
               rank <span class="op">=</span> <span class="st">"Phylum"</span>,
               detections <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="fl">0.1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">5</span>, <span class="fl">10</span>, <span class="fl">20</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>,
               prevalences <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">1</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-11-1.png" width="3000"></p>
</div>
<div id="tree-plotting" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-plotting" class="anchor"></a>Tree plotting</h1>
<p>The information stored in the <code>rowTree</code> can be directly plotted. However, sizes of stored trees have to be kept in mind and plotting of large trees rarely makes sense.</p>
<p>For this example we limit the information plotted to the top 100 taxa as judged by mean abundance on the genus level.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/FelixErnst/mia">mia</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/agglomerate-methods.html">agglomerateByRank</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span>
<span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html">addPerFeatureQC</a></span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span>
<span class="fu">rowData</span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">log_mean</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">mean</span><span class="op">)</span>
<span class="fu">rowData</span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span> <span class="op">&lt;-</span> 
    <span class="fu">rowData</span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span> <span class="op">/</span> <span class="fl">100</span>
<span class="va">top_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/getTopTaxa.html">getTopTaxa</a></span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span>,
                       method<span class="op">=</span><span class="st">"mean"</span>,
                       top<span class="op">=</span><span class="fl">100L</span>,
                       abund_values<span class="op">=</span><span class="st">"counts"</span><span class="op">)</span></code></pre></div>
<p>Colour, size and shape of tree tips and nodes can be decorated based on data present in the <code>SE</code> object or by providing additional information via the <code>other_fields</code> argument. Note that currently information for nodes have to be provided via the <code>other_fields</code> arguments.</p>
<p>Data will be matched via the <code>node</code> or <code>label</code> argument depending on which was provided. <code>label</code> takes precedent.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,
            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span>,
            tip_size_by <span class="op">=</span> <span class="st">"detected"</span><span class="op">)</span>
<span class="co">#&gt; Warning in .numeric_ij(ij = i, x = x, dim = "row"): For rows/cols with the same</span>
<span class="co">#&gt; name, only one is output</span>
<span class="co">#&gt; Warning in convertNode(tree = track, node = oldAlias): Multiple nodes are found</span>
<span class="co">#&gt; to have the same label.</span>
<span class="co">#&gt; Warning in convertNode(tree = value, node = nlk$nodeLab[iRep]): Multiple nodes</span>
<span class="co">#&gt; are found to have the same label.</span>
<span class="co">#&gt; Warning: `mutate_()` was deprecated in dplyr 0.7.0.</span>
<span class="co">#&gt; Please use `mutate()` instead.</span>
<span class="co">#&gt; See vignette('programming') for more help</span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot1-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size)" width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size)
</p>
</div>
<p>Tip and node labels can be shown as well. Setting <code>show_label = TRUE</code> shows the tip labels only …</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,
            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span>,
            tip_size_by <span class="op">=</span> <span class="st">"detected"</span>,
            show_label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Warning in .numeric_ij(ij = i, x = x, dim = "row"): For rows/cols with the same</span>
<span class="co">#&gt; name, only one is output</span>
<span class="co">#&gt; Warning in convertNode(tree = track, node = oldAlias): Multiple nodes are found</span>
<span class="co">#&gt; to have the same label.</span>
<span class="co">#&gt; Warning in convertNode(tree = value, node = nlk$nodeLab[iRep]): Multiple nodes</span>
<span class="co">#&gt; are found to have the same label.</span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot2-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size). Tip labels of the tree are shown as well." width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size). Tip labels of the tree are shown as well.
</p>
</div>
<p>… whereas node labels can be selectively shown by providing a named logical vector to <code>show_label</code>.</p>
<p>Please not that currently <code>ggtree</code> only can plot node labels in a rectangular layout.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Genus:Providencia"</span>, <span class="st">"Genus:Morganella"</span>, <span class="st">"0.961.60"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,
            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span>,
            tip_size_by <span class="op">=</span> <span class="st">"detected"</span>,
            show_label <span class="op">=</span> <span class="va">labels</span>,
            layout<span class="op">=</span><span class="st">"rectangular"</span><span class="op">)</span>
<span class="co">#&gt; Warning in .numeric_ij(ij = i, x = x, dim = "row"): For rows/cols with the same</span>
<span class="co">#&gt; name, only one is output</span>
<span class="co">#&gt; Warning in convertNode(tree = track, node = oldAlias): Multiple nodes are found</span>
<span class="co">#&gt; to have the same label.</span>
<span class="co">#&gt; Warning in convertNode(tree = value, node = nlk$nodeLab[iRep]): Multiple nodes</span>
<span class="co">#&gt; are found to have the same label.</span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot3-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size). Selected node and tip labels are shown." width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance (colour) and prevalence (size). Selected node and tip labels are shown.
</p>
</div>
<p>Information can also be visualized on the edges of the tree plot.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTree.html">plotRowTree</a></span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>,<span class="op">]</span>,
            edge_colour_by <span class="op">=</span> <span class="st">"Phylum"</span>,
            tip_colour_by <span class="op">=</span> <span class="st">"log_mean"</span><span class="op">)</span>
<span class="co">#&gt; Warning in .numeric_ij(ij = i, x = x, dim = "row"): For rows/cols with the same</span>
<span class="co">#&gt; name, only one is output</span>
<span class="co">#&gt; Warning in convertNode(tree = track, node = oldAlias): Multiple nodes are found</span>
<span class="co">#&gt; to have the same label.</span>
<span class="co">#&gt; Warning in convertNode(tree = value, node = nlk$nodeLab[iRep]): Multiple nodes</span>
<span class="co">#&gt; are found to have the same label.</span></code></pre></div>
<div class="figure">
<img src="miaViz_files/figure-html/plot4-1.png" alt="Tree plot using ggtree with tip labels decorated by mean abundance (colour) and edges labeled Kingdom (colour) and prevalence (size)" width="3000"><p class="caption">
Tree plot using ggtree with tip labels decorated by mean abundance (colour) and edges labeled Kingdom (colour) and prevalence (size)
</p>
</div>
</div>
<div id="graph-plotting" class="section level1">
<h1 class="hasAnchor">
<a href="#graph-plotting" class="anchor"></a>Graph plotting</h1>
<p>Similar to tree data, graph data can also be plotted in conjunction with <code>SummarizedExperiment</code> objects. Since the graph data in itself cannot be stored in a specialized slot, a graph object can be provided separately or as an element from the <code>metedata</code>.</p>
<p>Here we load an example graph. As graph data, all objects types accepted by <code>as_tbl_graph</code> from the <code>tidygraph</code> package are supported.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">col_graph</span><span class="op">)</span></code></pre></div>
<p>In the following examples, the <code>weight</code> data is automatically generated from the graph data. The <code>SummarizedExperiment</code> provided is required to have overlapping rownames with the node names of the graph. Using this link the graph plot can incorporated data from the <code>SummarizedExperiment</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotGraph.html">plotColGraph</a></span><span class="op">(</span><span class="va">col_graph</span>,
             <span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span>,
             colour_by <span class="op">=</span> <span class="st">"SampleType"</span>,
             edge_colour_by <span class="op">=</span> <span class="st">"weight"</span>,
             edge_width_by <span class="op">=</span> <span class="st">"weight"</span>,
             show_label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-15-1.png" width="3000"></p>
<p>As mentioned the graph data can be provided from the <code>metadata</code> of the <code>SummarizedExperiment</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">metadata</span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">GlobalPatterns</span>,<span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">graph</span> <span class="op">&lt;-</span> <span class="va">col_graph</span></code></pre></div>
<p>This produces the same plot as shown above.</p>
</div>
<div id="plotting-of-serial-data" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-of-serial-data" class="anchor"></a>Plotting of serial data</h1>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbiomeDataSets</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: MultiAssayExperiment</span>
<span class="va">silverman</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiomeDataSets/man/SilvermanAGutData.html">SilvermanAGutData</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; using temporary cache /tmp/RtmpsrcI0n/BiocFileCache</span>
<span class="co">#&gt; snapshotDate(): 2021-03-18</span>
<span class="co">#&gt; see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; see ?microbiomeDataSets and browseVignettes('microbiomeDataSets') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="va">silverman</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformCounts.html">transformCounts</a></span><span class="op">(</span><span class="va">silverman</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span>
<span class="va">taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/getTopTaxa.html">getTopTaxa</a></span><span class="op">(</span><span class="va">silverman</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p>Data from samples collected along time can be visualized using <code>plotSeries</code>. The <code>x</code> argument is used to reference data from the <code>colData</code> to use as descriptor for ordering the data. The <code>y</code> argument selects the feature to show. Since plotting a lot of features is not advised a maximum of 20 features can plotted at the same time.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSeries.html">plotSeries</a></span><span class="op">(</span><span class="va">silverman</span>,
           x <span class="op">=</span> <span class="st">"DAY_ORDER"</span>,
           y <span class="op">=</span> <span class="va">taxa</span>,
           colour_by <span class="op">=</span> <span class="st">"Family"</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-19-1.png" width="3000"> If replicated data is present, data is automatically used for calculation of the <code>mean</code> and <code>sd</code> and plotted as a range. Data from different assays can be used for plotting via the <code>abund_values</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSeries.html">plotSeries</a></span><span class="op">(</span><span class="va">silverman</span><span class="op">[</span><span class="va">taxa</span>,<span class="op">]</span>,
           x <span class="op">=</span> <span class="st">"DAY_ORDER"</span>,
           colour_by <span class="op">=</span> <span class="st">"Family"</span>,
           linetype_by <span class="op">=</span> <span class="st">"Phylum"</span>,
           abund_values <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-20-1.png" width="3000"></p>
<p>Additional variables can be used to modify line type aesthetics.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSeries.html">plotSeries</a></span><span class="op">(</span><span class="va">silverman</span>,
           x <span class="op">=</span> <span class="st">"DAY_ORDER"</span>,
           y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/getTopTaxa.html">getTopTaxa</a></span><span class="op">(</span><span class="va">silverman</span>, <span class="fl">5</span><span class="op">)</span>,
           colour_by <span class="op">=</span> <span class="st">"Family"</span>,
           linetype_by <span class="op">=</span> <span class="st">"Phylum"</span>,
           abund_values <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></code></pre></div>
<p><img src="miaViz_files/figure-html/unnamed-chunk-21-1.png" width="3000"></p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2021-03-18 r80099)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.2 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] microbiomeDataSets_0.99.5        MultiAssayExperiment_1.17.20    </span>
<span class="co">#&gt;  [3] scater_1.19.11                   patchwork_1.1.1                 </span>
<span class="co">#&gt;  [5] miaViz_0.99.5                    ggraph_2.0.5                    </span>
<span class="co">#&gt;  [7] ggplot2_3.3.3                    mia_0.99.4                      </span>
<span class="co">#&gt;  [9] TreeSummarizedExperiment_1.99.11 Biostrings_2.59.2               </span>
<span class="co">#&gt; [11] XVector_0.31.1                   SingleCellExperiment_1.13.12    </span>
<span class="co">#&gt; [13] SummarizedExperiment_1.21.1      Biobase_2.51.0                  </span>
<span class="co">#&gt; [15] GenomicRanges_1.43.3             GenomeInfoDb_1.27.8             </span>
<span class="co">#&gt; [17] IRanges_2.25.6                   S4Vectors_0.29.9                </span>
<span class="co">#&gt; [19] BiocGenerics_0.37.1              MatrixGenerics_1.3.1            </span>
<span class="co">#&gt; [21] matrixStats_0.58.0               BiocStyle_2.19.1                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] AnnotationHub_2.23.2          BiocFileCache_1.15.1         </span>
<span class="co">#&gt;   [3] systemfonts_1.0.1             igraph_1.2.6                 </span>
<span class="co">#&gt;   [5] lazyeval_0.2.2                splines_4.1.0                </span>
<span class="co">#&gt;   [7] BiocParallel_1.25.5           digest_0.6.27                </span>
<span class="co">#&gt;   [9] htmltools_0.5.1.1             viridis_0.5.1                </span>
<span class="co">#&gt;  [11] fansi_0.4.2                   magrittr_2.0.1               </span>
<span class="co">#&gt;  [13] memoise_2.0.0                 ScaledMatrix_0.99.2          </span>
<span class="co">#&gt;  [15] cluster_2.1.1                 DECIPHER_2.19.2              </span>
<span class="co">#&gt;  [17] graphlayouts_0.7.1            pkgdown_1.6.1                </span>
<span class="co">#&gt;  [19] colorspace_2.0-0              rappdirs_0.3.3               </span>
<span class="co">#&gt;  [21] blob_1.2.1                    ggrepel_0.9.1                </span>
<span class="co">#&gt;  [23] textshaping_0.3.3             xfun_0.22                    </span>
<span class="co">#&gt;  [25] dplyr_1.0.5                   crayon_1.4.1                 </span>
<span class="co">#&gt;  [27] RCurl_1.98-1.3                jsonlite_1.7.2               </span>
<span class="co">#&gt;  [29] ape_5.4-1                     glue_1.4.2                   </span>
<span class="co">#&gt;  [31] polyclip_1.10-0               gtable_0.3.0                 </span>
<span class="co">#&gt;  [33] zlibbioc_1.37.0               DelayedArray_0.17.9          </span>
<span class="co">#&gt;  [35] BiocSingular_1.7.2            scales_1.1.1                 </span>
<span class="co">#&gt;  [37] DBI_1.1.1                     Rcpp_1.0.6                   </span>
<span class="co">#&gt;  [39] xtable_1.8-4                  viridisLite_0.3.0            </span>
<span class="co">#&gt;  [41] tidytree_0.3.3                bit_4.0.4                    </span>
<span class="co">#&gt;  [43] rsvd_1.0.3                    httr_1.4.2                   </span>
<span class="co">#&gt;  [45] RColorBrewer_1.1-2            ellipsis_0.3.1               </span>
<span class="co">#&gt;  [47] pkgconfig_2.0.3               farver_2.1.0                 </span>
<span class="co">#&gt;  [49] scuttle_1.1.18                dbplyr_2.1.0                 </span>
<span class="co">#&gt;  [51] sass_0.3.1                    utf8_1.2.1                   </span>
<span class="co">#&gt;  [53] AnnotationDbi_1.53.1          later_1.1.0.1                </span>
<span class="co">#&gt;  [55] tidyselect_1.1.0              labeling_0.4.2               </span>
<span class="co">#&gt;  [57] rlang_0.4.10                  munsell_0.5.0                </span>
<span class="co">#&gt;  [59] BiocVersion_3.13.1            tools_4.1.0                  </span>
<span class="co">#&gt;  [61] cachem_1.0.4                  DirichletMultinomial_1.33.2  </span>
<span class="co">#&gt;  [63] generics_0.1.0                RSQLite_2.2.4                </span>
<span class="co">#&gt;  [65] ExperimentHub_1.17.1          evaluate_0.14                </span>
<span class="co">#&gt;  [67] stringr_1.4.0                 fastmap_1.1.0                </span>
<span class="co">#&gt;  [69] yaml_2.2.1                    ragg_1.1.2                   </span>
<span class="co">#&gt;  [71] ggtree_2.5.1                  knitr_1.31                   </span>
<span class="co">#&gt;  [73] bit64_4.0.5                   fs_1.5.0                     </span>
<span class="co">#&gt;  [75] tidygraph_1.2.0               purrr_0.3.4                  </span>
<span class="co">#&gt;  [77] KEGGREST_1.31.1               nlme_3.1-152                 </span>
<span class="co">#&gt;  [79] sparseMatrixStats_1.3.6       mime_0.10                    </span>
<span class="co">#&gt;  [81] aplot_0.0.6                   debugme_1.1.0                </span>
<span class="co">#&gt;  [83] compiler_4.1.0                png_0.1-7                    </span>
<span class="co">#&gt;  [85] interactiveDisplayBase_1.29.0 filelock_1.0.2               </span>
<span class="co">#&gt;  [87] curl_4.3                      beeswarm_0.3.1               </span>
<span class="co">#&gt;  [89] treeio_1.15.6                 tibble_3.1.0                 </span>
<span class="co">#&gt;  [91] tweenr_1.0.1                  bslib_0.2.4                  </span>
<span class="co">#&gt;  [93] stringi_1.5.3                 highr_0.8                    </span>
<span class="co">#&gt;  [95] desc_1.3.0                    lattice_0.20-41              </span>
<span class="co">#&gt;  [97] Matrix_1.3-2                  vegan_2.5-7                  </span>
<span class="co">#&gt;  [99] permute_0.9-5                 vctrs_0.3.6                  </span>
<span class="co">#&gt; [101] pillar_1.5.1                  lifecycle_1.0.0              </span>
<span class="co">#&gt; [103] BiocManager_1.30.10           jquerylib_0.1.3              </span>
<span class="co">#&gt; [105] BiocNeighbors_1.9.4           cowplot_1.1.1                </span>
<span class="co">#&gt; [107] bitops_1.0-6                  irlba_2.3.3                  </span>
<span class="co">#&gt; [109] httpuv_1.5.5                  R6_2.5.0                     </span>
<span class="co">#&gt; [111] promises_1.2.0.1              bookdown_0.21                </span>
<span class="co">#&gt; [113] gridExtra_2.3                 vipor_0.4.5                  </span>
<span class="co">#&gt; [115] MASS_7.3-53.1                 assertthat_0.2.1             </span>
<span class="co">#&gt; [117] rprojroot_2.0.2               withr_2.4.1                  </span>
<span class="co">#&gt; [119] GenomeInfoDbData_1.2.4        mgcv_1.8-34                  </span>
<span class="co">#&gt; [121] grid_4.1.0                    beachmat_2.7.7               </span>
<span class="co">#&gt; [123] tidyr_1.1.3                   rmarkdown_2.7                </span>
<span class="co">#&gt; [125] DelayedMatrixStats_1.13.5     rvcheck_0.1.8                </span>
<span class="co">#&gt; [127] ggnewscale_0.4.5              ggforce_0.3.3                </span>
<span class="co">#&gt; [129] shiny_1.6.0                   ggbeeswarm_0.6.0</span></code></pre></div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Felix G.M. Ernst.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
